### DISCUSSION TITLE
Unable to open BIRT generated xls file in MS Excel

### DISCUSSION THREAD

**Discussion by grijeshkumarsingh-greytip** [eclipse-birt/birt#1506] [General]
I am using BIRT 4.4.2 with below render option for excel to generate the xls file.

EXCELRenderOption options = new EXCELRenderOption();
options.setWrappingText(false);
options.setOutputFormat("xls");

While opening the file I am getting the warning message as

The file format and extension of 'Statement.xls' don't match. Files may have been corrupted or not safe. Please do not open unless you trust the source. Do you still want to open it?

This problem prevents the file from being uploaded into the banking facilities to create bank transfers.

Please assist me in fixing this problem.

![image](https://github.com/eclipse-birt/birt/assets/126166314/15e2ee22-3216-4189-a9f8-abe42ff019c5)

**Comment by hvbtup** [comment]
Independent from the issue, 
you should use a new(er) BIRT release and you should no longer use the XLS format at all, because many companies consider it a security risk.
Better use XLSX instead.

**Comment by speckyspooky** [comment]
The version 4.4.2 includes the old excel-prototype of BIRT which produce xml-excel.
It is an xml-document which can be read from excel. The extention ".xls" was used in the past that excel can open it and the use can see which data type it is.

But at the end this file type wasn't never a normal file type of xls or xlsx. And excel verify the content and compare it with the extension.
Based on the differences on it you get this message. The only way is to swithc according to @hvbtup use a newer version of BIRT with the spudsoft-excel and then you can use the file format of XLSX.

XLSX is also the better version because some features of BIRT for excel will only supported on XLSX and not longer for XLS.

**Comment by grijeshkumarsingh-greytip** [comment]
@hvbtup 
@speckyspooky 

Thanks a lot for your suggestions.
I would like to know which version of Birt is stable with Java 8 and Java 11 and supports XLSX with spudsoft-excel.

**Comment by hvbtup** [comment]
AFAIK you can use the Spudsoft excel emitter with 4.4.2; it's just a matter of configuration.
We were using the original Spudsoft emitter with 4.3.0; it can generate files in the XLS and XLSX formats.

What we did was basically:

	IRenderOption options = new RenderOption();
	options.setOutputFormat("XLS_SPUDSOFT");

The software was available at bitbucket, but it disappeared a long time ago.

I won't help you more than this about how to configure BIRT 4.4.2 with the Spudsoft emitter.
Instead I encourage you to make your software ready for Java 17 and BIRT 4.14 instead of investing time in solving a mere annoyance of ages-old software.

Remember that the Excel "alert" is really just a warning; I'm pretty confident that Excel will show the content if you click "Yes".
As a workaround, before uploading, you can open the file in Excel and save it as XLSX.
Of course this is a manual task.

The only Excel file format that is "future-proof" for the near future is XLSX.

**Comment by grijeshkumarsingh-greytip** [comment]
@hvbtup 

Thanks a lot for your suggestions.
Now I am able to generate the report in xls format using below options

IRenderOption options = new RenderOption();
options.setOutputFormat("XLS_SPUDSOFT");

But I am facing below error / warning while generating the report which created with report.rpttemplate template and report generating 
only with header.

Dec 09, 2023 2:49:47 PM uk.co.spudsoft.birt.emitters.excel.framework.Logger warn
WARNING: Not visiting class org.eclipse.birt.report.engine.content.impl.TableContent.

<img width="782" alt="birt_excel" src="https://github.com/eclipse-birt/birt/assets/126166314/1b61a002-81cd-4f85-b5bd-35fdb90d5bcf">

**Reply by hvbtup** [reply]
Sorry, I can't help regarding this issue.

**Reply by grijeshkumarsingh-greytip** [reply]
@hvbtup 
Now issue is resolved.
Issue is due to table permission.
Thanks @hvbtup @speckyspooky for your support.
